src/sentry/tasks/statistical_detectors.py

```diff
@@ -68,6 +69,39 @@
 TIMESERIES_PER_BATCH = 10
 
 
+def get_performance_project_settings(project_ids: List[int]):
+    project_settings = {}
+    project_option_settings = ProjectOption.objects.get_value_bulk(
+        project_ids, "sentry:performance_issue_settings"
+    )
+
+    for project_id in project_ids:
+        default_project_settings = projectoptions.get_well_known_default(
+            "sentry:performance_issue_settings",
+            project=project_id,
+        )
+
+        project_settings[project_id] = {
+            **default_project_settings,
+            **project_option_settings[project_id],
+        }  # Merge saved project settings into default so updating the default to add new settings works in the future.
+
+    return project_settings
+
+
+def all_projects_with_settings():
+    for projects in chunked(
+        RangeQuerySetWrapper(
+            Project.objects.filter(status=ObjectStatus.ACTIVE).select_related("organization"),
+            step=100,
+        ),
+        100,
+    ):
+        project_settings = get_performance_project_settings([project.id for project in projects])
+        for project, project_id in zip(projects, project_settings):
```

Comment:
Since `get_performance_project_settings` is returning a mapping from project id to the settings, there's no reason to zip them here. You can just do
```python
for project in projects:
    yield project, project_settings[project.id]
```

Reply:
done!